### ISSUE TITLE
bastion rdp using azure cli

### ISSUE THREAD

**Issue by mfmalto** [Azure/azure-cli#23965]
## Describe the bug

**Command Name**
`az network bastion rdp`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
cannot import name 'WinDLL' from 'ctypes' (/usr/lib/python3.9/ctypes/__init__.py)
Traceback (most recent call last):
  File "/usr/lib64/az/lib/python3.9/site-packages/knack/cli.py", line 233, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 663, in execute
    raise ex
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 697, in _run_job
    result = cmd_copy(params)
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 333, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/command_modules/network/custom.py", line 8396, in rdp_bastion_host
    from ._process_helper import launch_and_wait
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/command_modules/network/_process_helper.py", line 9, in <module>
    from ctypes import WinDLL, c_int, c_size_t, Structure, WinError, sizeof, pointer
ImportError: cannot import name 'WinDLL' from 'ctypes' (/usr/lib/python3.9/ctypes/__init__.py)

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az network bastion rdp --name {} --resource-group {} --target-resource-id {}`

## Environment Summary
```
Linux-5.4.0-1090-azure-x86_64-with-glibc2.35 (Cloud Shell), CBL-Mariner/Linux
Python 3.9.13
Installer: RPM

azure-cli 2.40.0

Extensions:
ai-examples 0.2.5
ml 2.7.1
ssh 1.1.2

Dependencies:
msal 1.18.0b1
azure-mgmt-resource 21.1.0b1
```

**Comment by yonzhan** [comment]
route to CXP team

**Comment by navba-MSFT** [comment]
@mfmalto Thanks for reaching out to us and reporting this issue. We are looking into this issue and we will provide an update.

**Comment by navba-MSFT** [comment]
@mfmalto Could you please confirm if you are running this from a cloudshell on Linux ? If Yes, it wont be able to open the RDP on your local machine. For this reason, we recommend using powershell with az cli (on **Windows** machine).

Also, note that RDP command works best on Windows, for other OS’ (like **Linux**) you can do the [**tunnel command**](https://docs.microsoft.com/en-us/azure/bastion/connect-native-client-windows#connect-tunnel) and then use the client of your choice.

>Connect to a VM using the [az network bastion tunnel command](https://docs.microsoft.com/en-us/azure/bastion/connect-native-client-windows#connect-tunnel). This lets you do the following:
>- Use native clients on non-Windows local computers (example: a Linux PC).
>- Use the native client of your choice. (This includes the Windows native client.)
>- Connect using SSH or RDP.
>- Set up concurrent VM sessions with Bastion.
>- [Upload files](https://docs.microsoft.com/en-us/azure/bastion/vm-upload-download-native#tunnel-command) to your target VM from your local computer. File download from the target VM to the local client is currently not supported for this command.
>
>Currently, this feature has the following limitation:
>
Signing in using an SSH private key stored in Azure Key Vault isn’t supported with this feature. Before signing in to your Linux VM using an SSH key pair, download your private key to a file on your local machine.

Hope this helps.

**Comment by navba-MSFT** [comment]
@mfmalto Thanks for your reply. I shall wait for the updates from you.

**Comment by mfmalto** [comment]
$BastionName='Bastion'
$BastionRG='AVD-POC-BASTION-RG'
$VMResourceID='/subscriptions/<sub-id>/resourceGroups/AVD-POC-MM/providers/Microsoft.Compute/virtualMachines/avd-poc-ss-mm'

az network bastion rdp --name $BastionName --resource-group $BastionRG --target-resource-id $VMResourceID

I’ve ever tried not using the vars and putting everything in line.

I get this error from PS Admin or from PS within terminal:
'rdp' is misspelled or not recognized by the system.

If I run it from the windows terminal using CLI, that’s when I get the other error.

Thanks!

**Comment by mfmalto** [comment]
I ran az upgrade in my powershell window and now get this when I run:

Command group 'network bastion' is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
Request to EncodingReservedUnitTypes v2 API endpoint failed.

I also updated to powershell 7 and get the same error.

**Comment by navba-MSFT** [comment]
@mfmalto Thanks for getting back. I see that you are now getting the below error:

>**Request to EncodingReservedUnitTypes v2 API endpoint failed.**

If you are getting this error, You need to open the bastion setting ---> _Enable the Native Client support_ (as highlighted in below screenshot):

![image](https://user-images.githubusercontent.com/57353862/191890481-11e95832-95d2-4e5f-bfc5-8e5087bd9565.png)

**Comment by navba-MSFT** [comment]
@mfmalto I wanted to do quick follow-up to check if you had a chance to look at my above comment. Please let us know if you had any updates on this. Awaiting your reply.

**Comment by navba-MSFT** [comment]
@mfmalto Thanks for getting back. I am glad that it connects now. May I know what is the error that you are getting while authenticating ? Please share the detailed error message here (with screenshot). Awaiting your reply.

I am sharing a few similar issues reported in QnA forum to check if that helps you:
https://learn.microsoft.com/en-us/answers/questions/409639/enable-azure-ad-login-with-bastion-on-exisitng-vm.html
https://learn.microsoft.com/en-us/answers/questions/391362/unable-to-login-to-azure-vm.html
https://learn.microsoft.com/en-us/azure/active-directory/devices/howto-vm-sign-in-azure-ad-windows#log-in-by-using-azure-ad-credentials-to-a-windows-vm
https://learn.microsoft.com/en-us/azure/active-directory/devices/howto-vm-sign-in-azure-ad-linux

**Comment by navba-MSFT** [comment]
@mfmalto I wanted to do quick follow-up to check if you had a chance to look at my above comment. Please let us know if you had any updates on this. Awaiting your reply.

**Comment by navba-MSFT** [comment]
@mfmalto The action is currently pending on you to share the error details that you are getting while authenticating ? Please feel  free to reopen this thread once you have these details. We would be happy to help.

**Comment by jiasli** [comment]
Duplicate of #23143
